src/Campaigns/Migrations/MigrateFormsToCampaignForms.php

```diff
@@ -29,66 +26,118 @@ public static function id(): string
-            'logo' => $form->settings->designSettingsLogoUrl,
-            'image' => $form->settings->designSettingsImageUrl,
-            'primaryColor' => $form->settings->primaryColor,
-            'secondaryColor' => $form->settings->secondaryColor,
-            'goal' => (int) $form->settings->goalAmount,
-            'status' => $this->mapFormStatusToCampaignStatus($form->status),
-            'startDate' => new DateTime($form->settings->goalStartDate),
-            'endDate' => new DateTime($form->settings->goalEndDate),
-        ]);
+        $formId = $formData->id;
+        $formTitle = $formData->title;
+        $formStatus = $formData->status;
+        $formCreatedAt = $formData->createdAt;
+        $formSettings = json_decode($formData->settings);
+
+        $campaignId = DB::table('give_campaigns')
+            ->insert([
+                'campaign_type' => 'core',
+                'campaign_title' => $formTitle,
+                'status' => $this->mapFormToCampaignStatus($formStatus),
+                'short_desc' => $formSettings->formExcerpt,
+                'long_desc' => $formSettings->description,
+                'campaign_logo' => $formSettings->designSettingsLogoUrl,
+                'campaign_image' => $formSettings->designSettingsImageUrl,
+                'primary_color' => $formSettings->primaryColor,
+                'secondary_color' => $formSettings->secondaryColor,
+                'campaign_goal' => $formSettings->goalAmount,
+                'start_date' => $formSettings->goalStartDate,
+                'end_date' => $formSettings->goalEndDate,
+                'date_created' => $formCreatedAt,
+            ]);
 
         DB::table('give_campaign_forms')
             ->insert([
-                'form_id' => $form->id,
-                'campaign_id' => $campaign->id,
+                'form_id' => $formId,
+                'campaign_id' => $campaignId,
             ]);
     }
 
-   /**
+    /**
      * @unreleased
      */
-    public function mapFormStatusToCampaignStatus(DonationFormStatus $status)
+    public function mapFormToCampaignStatus(string $status): string
     {
         switch ($status) {
-            case DonationFormStatus::PUBLISHED():
-            case DonationFormStatus::UPGRADED(): // TODO: How do we handle upgraded, non-upgraded forms?
-            case DonationFormStatus::PRIVATE(): // TODO: How do we handle Private forms?
-                return CampaignStatus::ACTIVE();
 
-            case DonationFormStatus::PENDING():
-                return CampaignStatus::PENDING();
+            case 'pending':
+                return 'pending';
+
+            case 'draft':
+                return 'draft';
+
+            case 'trash':
+                return 'inactive';
 
-            case DonationFormStatus::DRAFT():
-                return CampaignStatus::DRAFT();
+            case 'publish':
+            case 'private':
+            case 'upgraded':
```

Comment:
On second thought, I think the `upgraded` status only applied after a v2 form has been migrated to a v3 (new) form and the donations have been transferred, which means that forms with the `upgraded` status should probably be excluded from this migration.

Reply:
See [fe3f605](https://github.com/impress-org/givewp/pull/7532/commits/fe3f605c7c5424627bc9d9dec1156c5d128d47d2)